ibis/backends/dask/execution/util.py

```diff
@@ -90,3 +95,34 @@ def safe_concat(dfs: List[Union[dd.Series, dd.DataFrame]]) -> dd.DataFrame:
         result = dd.concat(dfs, axis=1)
 
     return result
+
+
+def compute_sort_key(key, data, timecontext, scope=None, **kwargs):
+    """
+    Note - we use this function instead of the pandas.execution.util so that
+    we use the dask `execute` method
+    """
+    by = key.to_expr()
+    name = ibis.util.guid()
+    try:
+        if isinstance(by, str):
+            return name, data[by]
+        return name, data[by.get_name()]
+    except com.ExpressionError:
+        if scope is None:
```

Comment:
why does this happen?

Reply:
When you pass an expression and we have to execute to get back a column to sort by,

Reply:
kk let's add a comment explaining this (can be next PR )